### PR TITLE
WIP projects:ad713x_fmcz: fix data read bug

### PR DESCRIPTION
**Pull request by adrimbarean**
The projct was returning always the same sample of data because the one
of the ADCs was used in a reserved mode of operation. The chip can only
function as a master on the data bus (generate SPI clock and ODR) or a
slave (accept SPI clock and ODR). The chip was used instead to generate
ODR, but accept the SPI clock which cannot work properly.

To accomodate this use a an AXI PWM generator to generate ODR at a
specified rate for both devices. Note: the ODR signal must be present
before chip initialization.

Signed-off-by: Andrei Drimbarean <Andrei.Drimbarean@analog.com>

PR is to be merged after the HDL changes propagate to master.

### CODE REVIEW

**Review by dbogdan — APPROVED** [review]

File: projects/ad713x_fmcz/src/ad713x_fmc.c [hunk 1]
```diff
@@ -72,18 +76,28 @@
...
 	struct ad713x_dev *ad713x_dev_2;
 	struct ad713x_init_param ad713x_init_param_1;
 	struct ad713x_init_param ad713x_init_param_2;
-	uint32_t *offload_data;
-	uint32_t i;
+	uint32_t i = 0, j;
 	int32_t ret;
+	const float lsb = 4.096 / (pow(2, 23));
+	float data;
+	uint32_t spi_eng_dma_flg = 6;
```
Comment (mchindri) [inline projects/ad713x_fmcz/src/ad713x_fmc.c:97]: Is there any define for 6?

**Review by mchindri — APPROVED** [review]

**Review by amiclaus — COMMENTED** [review]
`error: 'XPAR_ODR_GENERATOR_BASEADDR' undeclared (first use in this function); did you mean 'XPAR_DDR_MEM_BASEADDR'?`

`error: 'XPAR_AXI_AD7134_CLKGEN_BASEADDR' undeclared (first use in this function); did you mean 'XPAR_AXI_SDP_CLKGEN_BASEADDR'?`

Do we require a hdl update on the CI side for this?

**Comment by adrimbarean** [comment]
> Do we require a hdl update on the CI side for this?

Precisely. The work is done, just needs to be merged to master on the HDL side before we can merge this.

File: projects/ad713x_fmcz/src/ad713x_fmc.c (hunk 1 above)
Comment (adrimbarean) [reply projects/ad713x_fmcz/src/ad713x_fmc.c:97]: Will do.

**Review by amiclaus — APPROVED** [review]

**Comment by adrimbarean** [comment]
V1:
 - add partial reporting flag to AXI DMAC driver;
 - `uint32_t spi_eng_dma_flg = 6;` => `uint32_t spi_eng_dma_flg = DMA_LAST | DMA_PARTIAL_REPORTING_EN;`

**Comment by adrimbarean** [comment]
V2:
 - rebase on latest master

**Comment by adrimbarean** [comment]
V3:
 - include `axi_dmac.h` in main source to use DMA flags.

File: projects/ad713x_fmcz/src/parameters.h [hunk 2]
```diff
@@ -78,10 +78,6 @@
 #define AD7134_FMC_CH_NO		8
 #define AD7134_FMC_SAMPLE_NO		1024
 
-#define SPI_ENGINE_RX_ADDR		0x800000
```
Comment (amiclaus) [inline projects/ad713x_fmcz/src/parameters.h:81]: it looks like these defines are still used here :
https://github.com/analogdevicesinc/no-OS/blob/bfb5fecff0b18a7494bd0db0e7fd77467d8eae5e/projects/ad713x_fmcz/src/ad713x_fmc.c#L300-L302

**Review by amiclaus — CHANGES_REQUESTED** [review]
https://github.com/analogdevicesinc/no-OS/pull/1203/files#diff-90539af6fbd9d1894cf26ba395029f0900135ac1943c5b7af3829289d7d4778cL81-L83

it looks like these defines are still used here :
https://github.com/analogdevicesinc/no-OS/blob/bfb5fecff0b18a7494bd0db0e7fd77467d8eae5e/projects/ad713x_fmcz/src/ad713x_fmc.c#L300-L302

**Comment by adrimbarean** [comment]
V4:
 - fix IIO build;
 - rebase on latest master.

File: projects/ad713x_fmcz/.Xil/-20668-ADRIMBAR-L02/system.bda [hunk 3]
```diff
@@ -0,0 +1,357 @@
+﻿<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
```
Comment (amiclaus) [inline projects/ad713x_fmcz/.Xil/-20668-ADRIMBAR-L02/system.bda:1]: I believe this file was added by mistake 😄. Maybe we can add `.Xil` folder to gitignore.

File: drivers/axi_core/axi_dmac/axi_dmac.h [hunk 4]
```diff
@@ -78,7 +78,8 @@ enum dma_direction {
 
 enum dma_flags {
 	DMA_CYCLIC = 1,
-	DMA_LAST = 2
+	DMA_LAST = 2,
+	DMA_PARTIAL_REPORTING_EN = 4
```
Comment (amiclaus) [inline drivers/axi_core/axi_dmac/axi_dmac.h:82]: nit: I think the commit description `No functional changes` is not quite correct :)

**Review by amiclaus — CHANGES_REQUESTED** [review]

**Comment by adrimbarean** [comment]
V5:
 - removed unnecessary file;
 - add different commit description for the second commit.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**